src/message.rs

```diff
@@ -42,73 +42,96 @@ impl FromStr for AprsMessage {
     }
 }
 
+impl Display for AprsMessage {
+    fn fmt(&self, f: &mut Formatter) -> std::fmt::Result {
+        write!(
+            f,
+            ":{: <9}:{}",
+            String::from_utf8_lossy(&self.addressee),
+            String::from_utf8_lossy(&self.text)
+        )?;
+
+        if let Some(id) = &self.id {
+            write!(f, "{{{}", String::from_utf8_lossy(id))?;
+        }
+
+        Ok(())
+    }
+}
+
+fn remove_trailing_spaces(arr: &mut Vec<u8>) {
```

Comment:
I would probably change the name and type signature of this a little:

```suggestion
fn trim_spaces_end(arr: &[u8]) -> &[u8] {
```

Since this is effectively a variant on the `trim_*` methods from std.

Also worth noting that this is not quite equivalent to the original `addressee.trim()`, which trims all whitespace characters from both the ends. I'm not very familiar with APRS, but I'm assuming this was intentional and the APRS spec only cares about trailing space characters?

Reply:
You're correct - the APRS spec says that the addresses can have spaces at the end (but they should be ignored)